src/lib/storage/yaml_writer.rb

```diff
@@ -139,7 +158,7 @@ def yaml_partition(partition)
           "size" => DiskSize.new(partition.region.to_kb(partition.region.length)).to_s,
           "name" => partition.name,
           "type" => @partition_types[partition.type],
-          "id"   => @partition_ids[partition.id]
+          "id"   => @partition_ids[partition.id] || "0x#{partition.id.to_s(16)}"
```

Comment:
We have an enum for those IDs; that's what fdisk calls "partition type", as in 0x83==Linux or 0x82==swap etc.; symbolic names are very much preferred here for better readability. Hex numbers are also allowed, but we should really have an enum entry for it. But then, we can never know what some weird system is using that we don't have an enum value for yet, so it makes sense to have a hex fallback. Remember we might also use that to write probed values to disk, and there it might actually happen.